src/app/_components/RecentBlocks/useRecentBlocks.ts

```diff
@@ -0,0 +1,131 @@
+'use client';
+
+import { getBlocksV2ListQueryKey, useBlocksV2List } from '@/app/blocks/queries/useBlocksV2Queries';
+import { useHomePageData } from '@/app/context';
+import { getBurnBlocksQueryKey, useBurnBlocks } from '@/common/queries/useBurnBlocksInfinite';
+import { useQueryClient } from '@tanstack/react-query';
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+import { Block, NakamotoBlock } from '@stacks/stacks-blockchain-api-types';
+
+import { useGlobalContext } from '../../../common/context/useGlobalContext';
+import { useSubscribeBlocks } from '../BlockList/Sockets/useSubscribeBlocks';
+import { RECENT_BTC_BLOCKS_COUNT, RECENT_STX_BLOCKS_COUNT } from './consts';
+
+export type RecentBlocksType = 'btc' | 'stx';
+
+const isNewStxBlock = (block: NakamotoBlock | Block, lastBlockHeight: number | undefined) => {
+  return block.height > (lastBlockHeight || 0);
+};
+
+const isNewBtcBlock = (block: NakamotoBlock | Block, lastBlockHeight: number | undefined) => {
+  return block.burn_block_height > (lastBlockHeight || 0);
+};
+
+// Handle getting stx blocks data. If there is initial stx blocks data, set it in the cache and use it. Otherwise, query for the data
+const useStxBlocks = () => {
+  const { initialRecentBlocks } = useHomePageData();
+  const activeNetwork = useGlobalContext().activeNetwork;
+
+  const queryClient = useQueryClient();
+
+  const recentStxBlocks = initialRecentBlocks?.stxBlocks;
+  const isStxBlocksCacheSetWithInitialData = useRef(false);
+  if (isStxBlocksCacheSetWithInitialData.current === false && recentStxBlocks) {
+    const queryKey = getBlocksV2ListQueryKey(RECENT_STX_BLOCKS_COUNT, activeNetwork.url);
+    queryClient.setQueryData(queryKey, recentStxBlocks);
```

Comment:
Can we avoid running stQueryData inside render function? Maybe move it into a useEffect.